packages/models-library/src/models_library/projects_state.py

```diff
@@ -24,16 +24,24 @@ class RunningState(str, Enum):
     SEE StateType for task state
     """
 
-    UNKNOWN = "UNKNOWN"
-    PUBLISHED = "PUBLISHED"
-    NOT_STARTED = "NOT_STARTED"
-    PENDING = "PENDING"
-    WAITING_FOR_RESOURCES = "WAITING_FOR_RESOURCES"
-    STARTED = "STARTED"
-    SUCCESS = "SUCCESS"
-    FAILED = "FAILED"
-    ABORTED = "ABORTED"
-    WAITING_FOR_CLUSTER = "WAITING_FOR_CLUSTER"
+    UNKNOWN = "UNKNOWN"  # Error state: the task disappeared from the system (Dask backend lost the task and its state is unknown)
+    NOT_STARTED = "NOT_STARTED"  # Default state when the project is created
+
+    PUBLISHED = (
+        "PUBLISHED"  # The project is awaiting the compute scheduler to pick up the task
+    )
+
+    PENDING = "PENDING"  # The compute scheduler has taken over, and the task is waiting for a worker to pick it up
+
+    WAITING_FOR_CLUSTER = "WAITING_FOR_CLUSTER"  # No cluster is available to run the task; waiting for one to become available
+    WAITING_FOR_RESOURCES = "WAITING_FOR_RESOURCES"  # No worker is available to run the task; waiting for one to become available
+    # PENDING -> WAITING_FOR_CLUSTER -> PENDING -> WAITING_FOR_RESOURCES -> PENDING -> STARTED
+
+    STARTED = "STARTED"  # A worker has picked up the task and is executing it
+
+    SUCCESS = "SUCCESS"  # Task finished successfully
+    FAILED = "FAILED"  # Task finished with an error
+    ABORTED = "ABORTED"  # Task was aborted before completion
```

Comment:
this is mostly incorrect. I think I would recommend reverting this cause that does not make fully sense and will mostly confuse people that read that.

- UNKNOWN: that means the state is unknown. nothing to do with an error a priori
- NOT_STARTED: means that it is not started. whatever the RunningState is applied to
- PUBLISHED: means it is published. again applies to whatever the Runningstate is applied to
- PENDING: means it is pending, it can be in between any other state actually
- STARTED: means it is started
- FAILED: means it is in error

well I think best is to revert these comments.

Reply:
or perhaps add yours :-)